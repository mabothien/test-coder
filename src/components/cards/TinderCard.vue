<template>
  <v-container class="tinder_app">
    <v-row no-gutters class="justify-center">
      <v-col
        cols="6"
      >
        <v-card
          outlined
          tile
        >
          <v-tabs-items v-model="tab" class="tinder_content">
            <v-tab-item
              v-for="item in items"
              :key="item.tab"
            >
              <component
                :is="item.tab"
                :tab="tab"
                :like-list="likeList"
                :matches="matches"
                :random-user="randomUser"
                @reload="reloadDiscover()"
              />
            </v-tab-item>
          </v-tabs-items>
          <v-tabs
            v-model="tab"
            grow
          >
            <v-tab
              v-for="item in items"
              :key="item.tab"
              class="tinder_tab"
              @click="onTab(item.tab)"
            >
              <v-icon :color="item.tabIconColor">
                {{ item.tabIcon }}
              </v-icon> {{ item.tabName }}
            </v-tab>
          </v-tabs>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>
<script>
export default {
  data () {
    return {
      likeList: [],
      matches: [],
      randomUser: {},
      tab: 1,
      items: [
        {
          tab: 'TabLikeList', tabName: 'Liked', tabIconColor: 'red', tabIcon: 'mdi-emoticon-kiss-outline', content: 'Tab 1 Content'
        },
        {
          tab: 'TabDiscover', tabName: 'Discover', tabIconColor: 'green', tabIcon: 'mdi-heart', content: 'Tab 2 Content'
        },
        {
          tab: 'TabMatches', tabName: 'Matches', tabIconColor: 'blue', tabIcon: 'mdi-wechat', content: 'Tab 3 Content'
        }
      ]
    }
  },
  async mounted () {
    await Promise.all([
      this.fetchLikeList(),
      this.fetchMatchesList(),
      this.fetchDiscoverById()
    ])
  },
  methods: {
    async reloadDiscover () {
      await this.fetchDiscoverById()
    },
    // Call api from graplQL
    async fetchMatchesList () {
      try {
        const { data } = await this.$axios.request({
          url: 'http://localhost:4000/user',
          method: 'POST',
          data: {
            query: `
          {
            matchesUser {
              title,
              firstName,
              lastName,
              picture
            }
          }
         `
          }
        })
        this.matches = data.data.matchesUser
      } catch (error) {
        console.log(error)
      }
    },

    async fetchDiscoverById () {
      try {
        const { data } = await this.$axios.request({
          url: 'http://localhost:4000/user',
          method: 'POST',
          data: {
            query: ` 
          {
            randomUser {
              id,
              title,
              firstName,
              lastName,
              picture,
              isMatch,
              isPass
            }
          }
         `
          }
        })
        this.randomUser = data.data.randomUser
      } catch (error) {
        console.log(error)
      }
    },

    async fetchLikeList () {
      try {
        const { data } = await this.$axios.request({
          url: 'http://localhost:4000/user',
          method: 'POST',

          data: {
            query: `
          {
            likeUsers{
              title,
              firstName,
              lastName,
              picture
            }
          }
          `
          }
        })
        this.likeList = data.data.likeUsers
      } catch (error) {
        console.log(error)
      }
    },
    onTab (tab) {
      const index = this.items.findIndex(i => i.tab === tab)
      this.tab = index
      if (this.tab === 1) {
        this.fetchLikeList()
      } else if (this.tab === 2) {
        this.fetchMatchesList()
      } else {
        this.fetchDiscoverById()
      }
    }
  }
}
</script>
