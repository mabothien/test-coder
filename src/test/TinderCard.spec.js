import Vue from 'vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import TinderCard from '../components/cards/TinderCard'

Vue.use(Vuetify)
describe('TinderCard.vue', () => {
  const localVue = createLocalVue()
  let vuetify
  beforeEach(()=> {
    vuetify = new Vuetify()
  })

  it('should do something', () => {
    const wrapper = mount(TinderCard, {
      localVue,
      vuetify
    })
    const event = jest.fn()
    const button = wrapper.find('.v-tab')

    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    // wrapper.vm.$on('onTab:clicked', event)
    expect(event).toHaveBeenCalledTimes(0)

    // Simulate a click on the button
    button.trigger('click')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
  })
})