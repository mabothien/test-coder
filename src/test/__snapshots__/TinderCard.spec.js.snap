// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TinderCard.vue should do something 1`] = `
"<div class=\\"container tinder_app\\">
  <div class=\\"row justify-center no-gutters\\">
    <div class=\\"col col-6\\">
      <div class=\\"v-card v-sheet v-sheet--outlined theme--light rounded-0\\">
        <div class=\\"v-window tinder_content v-item-group theme--light v-tabs-items\\">
          <div class=\\"v-window__container\\">
            <transition-stub name=\\"\\">
              <!---->
            </transition-stub>
            <transition-stub name=\\"\\">
              <div class=\\"v-window-item v-window-item--active\\">
                <tabdiscover tab=\\"1\\" like-list=\\"\\" matches=\\"\\" random-user=\\"[object Object]\\"></tabdiscover>
              </div>
            </transition-stub>
            <transition-stub name=\\"\\">
              <!---->
            </transition-stub>
          </div>
        </div>
        <div class=\\"v-tabs v-tabs--grow theme--light\\">
          <div role=\\"tablist\\" class=\\"v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text\\">
            <div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\">
              <transition-stub name=\\"fade-transition\\"></transition-stub>
            </div>
            <div class=\\"v-slide-group__wrapper\\">
              <div class=\\"v-slide-group__content v-tabs-bar__content\\">
                <div class=\\"v-tabs-slider-wrapper\\" style=\\"transition: none;\\">
                  <div class=\\"v-tabs-slider\\"></div>
                </div>
                <div tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"tinder_tab v-tab\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-emoticon-kiss-outline theme--light red--text\\"></i> Liked
                </div>
                <div tabindex=\\"0\\" aria-selected=\\"true\\" role=\\"tab\\" class=\\"tinder_tab v-tab v-tab--active\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-heart theme--light green--text\\"></i> Discover
                </div>
                <div tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"tinder_tab v-tab\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-wechat theme--light blue--text\\"></i> Matches
                </div>
              </div>
            </div>
            <div class=\\"v-slide-group__next v-slide-group__next--disabled\\">
              <transition-stub name=\\"fade-transition\\"></transition-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
